chalice/deploy/packager.py

```diff
@@ -1,116 +1,142 @@
 
-    def _create_virtualenv(self, venv_dir):
-        # type: (str) -> None
-        virtualenv.create_environment(venv_dir)
+    def __init__(self, osutils=None, dependency_builder=None):
+        # type: (Optional[OSUtils], Optional[DependencyBuilder]) -> None
+        if osutils is None:
+            osutils = OSUtils()
+        self._osutils = osutils
+        if dependency_builder is None:
+            dependency_builder = DependencyBuilder(self._osutils)
+        self._dependency_builder = dependency_builder
 
-    def create_deployment_package(self, project_dir, package_filename=None):
-        # type: (str, Optional[str]) -> str
+    def _get_requirements_filename(self, project_dir):
+        # type: (str) -> str
+        # Gets the path to a requirements.txt file out of a project dir path
+        return self._osutils.joinpath(project_dir, 'requirements.txt')
+
+    def create_deployment_package(self, project_dir, python_version,
+                                  package_filename=None):
+        # type: (str, str, Optional[str]) -> str
         print("Creating deployment package.")
-        # pip install -t doesn't work out of the box with homebrew and
-        # python, so we're using virtualenvs instead which works in
-        # more cases.
-        venv_dir = os.path.join(project_dir, '.chalice', 'venv')
-        self._create_virtualenv(venv_dir)
-        pip_exe = compat.pip_script_in_venv(venv_dir)
-        assert os.path.isfile(pip_exe)
-        # Next install any requirements specified by the app.
-        requirements_file = os.path.join(project_dir, 'requirements.txt')
+        # Now we need to create a zip file and add in the site-packages
+        # dir first, followed by the app_dir contents next.
         deployment_package_filename = self.deployment_package_filename(
-            project_dir)
+            project_dir, python_version)
         if package_filename is None:
             package_filename = deployment_package_filename
-        if self._has_at_least_one_package(requirements_file) and not \
-                os.path.isfile(package_filename):
-            p = subprocess.Popen([pip_exe, 'install', '-r', requirements_file],
-                                 stdout=subprocess.PIPE)
-            p.communicate()
-        deps_dir = compat.site_packages_dir_in_venv(venv_dir)
-        assert os.path.isdir(deps_dir)
-        # Now we need to create a zip file and add in the site-packages
-        # dir first, followed by the app_dir contents next.
-        dirname = os.path.dirname(os.path.abspath(package_filename))
-        if not os.path.isdir(dirname):
-            os.makedirs(dirname)
-        with zipfile.ZipFile(package_filename, 'w',
-                             compression=zipfile.ZIP_DEFLATED) as z:
-            self._add_py_deps(z, deps_dir)
+        try:
+            requirements_filepath = self._get_requirements_filename(
+                project_dir)
+            site_packages_dir = self._osutils.joinpath(
```

Comment:
Yep thats the plan. Ill just make the debug hook copy it out somewhere before it gets erased.